src/Seq.Input.HealthCheck/HttpHealthCheck.cs

```diff
@@ -40,15 +43,32 @@ class HttpHealthCheck
         const int InitialContentChars = 16;
         const string OutcomeSucceeded = "succeeded", OutcomeFailed = "failed";
 
-        public HttpHealthCheck(HttpClient httpClient, string title, string targetUrl, JsonDataExtractor extractor, bool bypassHttpCaching)
+        public HttpHealthCheck(HttpClient httpClient, string title, string targetUrl, string authenticationHeaderValue, string optionalHeader, JsonDataExtractor extractor, bool bypassHttpCaching)
         {
             _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
             _title = title ?? throw new ArgumentNullException(nameof(title));
             _targetUrl = targetUrl ?? throw new ArgumentNullException(nameof(targetUrl));
+            _authenticationHeaderValue = authenticationHeaderValue;
+            _optionalHeader = optionalHeader;
             _extractor = extractor;
             _bypassHttpCaching = bypassHttpCaching;
         }
 
+        private static void AddOrOverwriteHeader(HttpRequestMessage request, string attribute, string value)
+        {
+            // opportunistic abort: header values cannot be empty (https://datatracker.ietf.org/doc/html/rfc7230#section-3.2)
+            if (string.IsNullOrWhiteSpace(value))
```

Comment:
It's not 100% clear that this is the case, there are rather a lot of threads suggesting that empty field values are acceptable; in the linked EBNF the rule:

```ebnf
field-value    = *( field-content / obs-fold )
```

has a 0-or-more `*` so may be better to just let these through.

Reply:
You are right, the API does allow _empty_ values, but not _null_ values. I'll add that.